zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/incident/CallActivityIncidentTest.java

```diff
@@ -85,6 +86,41 @@ public void shouldCreateIncidentIfProcessIsNotDeployed() {
                 + "' to be deployed, but not found.");
   }
 
+  @Test
+  public void shouldCreateIncidentIfProcessIsNotDeployedInSameDeploymentForBindingTypeDeployment() {
+    // given
+    final var childProcessId = Strings.newRandomValidBpmnId();
+    final var childProcess = Bpmn.createExecutableProcess(childProcessId).startEvent().done();
+    ENGINE.deployment().withXmlResource("wf-child.bpmn", childProcess).deploy();
+    final var parentProcess =
+        Bpmn.createExecutableProcess(parentProcessId)
+            .startEvent()
+            .callActivity(
+                "call",
+                c -> c.zeebeProcessId(childProcessId).zeebeBindingType(ZeebeBindingType.deployment))
```

Comment:
Oh, you nicely anticipated the upcoming changes here 😎. I would keep the test as is though since it is reflecting the current behavior and adapt it later as part of #19712.